reference-data-proxy/src/v1/controllers/estore/eStore.controller.ts

```diff
@@ -1,84 +1,335 @@
+      await cronJobLogsCollection.updateOne(
+        { dealIdentifier: eStoreData.dealIdentifier, exporterName: eStoreData.exporterName, buyerName: eStoreData.buyerName },
+        { $set: { siteCronJob: { status: ESTORE_CRON_STATUS.PENDING } } },
+      );
+
+      // send a request to eStore to start creating the eStore site
+      console.info('API Call started: Create a new eStore site for ', eStoreData.exporterName);
+      const siteCreationResponse = await createExporterSite({ exporterName: eStoreData.exporterName });
+
+      // check if the siteCreation endpoint returns a siteName - this is usually a number (i.e. 12345)
+      if (siteCreationResponse?.data?.siteName) {
+        // update the database with the new siteName
+        await cronJobLogsCollection.updateOne(
+          { dealIdentifier: eStoreData.dealIdentifier, exporterName: eStoreData.exporterName, buyerName: eStoreData.buyerName },
+          { $set: { siteName: siteCreationResponse.data.siteName } },
+        );
+        // add a new job to the `Cron Job Manager` queue that runs every 50 seconds
+        // in general, the site creation should take around 4 minutes, but we can check regularly to see if the site was created
+        eStoreCronJobManager.add(siteCreationResponse.data.siteName, siteCreationTimer, async () => {
+          await eStoreSiteCreationJob(eStoreData);
+        });
+        console.info('Cron job started: eStore Site Creation Cron Job started ', siteCreationResponse.data.siteName);
+        // update the database to indicate that the `site cron job` started
+        await cronJobLogsCollection.updateOne(
+          { dealIdentifier: eStoreData.dealIdentifier, exporterName: eStoreData.exporterName, buyerName: eStoreData.buyerName },
+          { $set: { 'siteCronJob.status': ESTORE_CRON_STATUS.RUNNING, 'siteCronJob.startDate': Date.now() } },
+        );
+        eStoreCronJobManager.start(siteCreationResponse.data.siteName);
+      } else {
+        console.error('API Call failed: Unable to create a new site in eStore', { siteCreationResponse });
+        // update the database to indicate that the API call failed
+        await cronJobLogsCollection.updateOne(
+          { dealIdentifier: eStoreData.dealIdentifier, exporterName: eStoreData.exporterName, buyerName: eStoreData.buyerName },
+          { $set: siteCreationResponse },
+        );
+      }
+    } else {
+      console.error('API Call failed: Unable to check if a site exists', { siteExistsResponse });
+      // update the database to indicate that the API call failed
+      await cronJobLogsCollection.updateOne(
+        { dealIdentifier: eStoreData.dealIdentifier, exporterName: eStoreData.exporterName, buyerName: eStoreData.buyerName },
+        { $set: { siteExistsResponse } },
+      );
+    }
+  } else {
+    console.error('eStore body is empty', { eStoreData });
+    res.status(400).send({ message: 'eStore body is empty', status: 400 });
+  }
```

Comment:
I would usually say it's worth splitting up more functions into different files so it's not one big file and less overwhelming to look at, but for estore maybe it's better to keep in one file? Especially if going to walk through this with estore team etc.

Reply:
I'd like to send them the entire file. I agree though, it could be shorter than this